tests/test_responses_between_dates.py

```diff
@@ -0,0 +1,97 @@
+            {"respondent_id": 2, "date": "1970-01-16", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 2, "answer": "Generous Giver"}, 
+            ]),
+
+            #Date range 1 year prior to the date where test data is stored
+            ({"start_date": "1969-01-01","end_date": "1969-12-31"},[]),
+
+            #Current date
+            ({"start_date": str(date.today()),"end_date": str(date.today())},[]),
+
+            #Date range covers first week of test data
+            ({"start_date": "1969-12-29","end_date": "1970-01-02"},[
+            {"respondent_id": 1, "date": "1970-01-02", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 2, "answer": "Generous Giver"}
+            ]),
+
+            #Date range covers first two weeks of test data
+            ({"start_date": "1969-12-29","end_date": "1970-01-09"},[
+            {"respondent_id": 1, "date": "1970-01-02", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 2, "answer": "Generous Giver"},
+            {"respondent_id": 1, "date": "1970-01-09", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-09", "question_no": 1, "answer": "Generous Giver"}
+            ]),
+
+            #Date range from a week prior to the dates where test data is present ,to the week succeeding the last date which has the test date
+            ({"start_date": "1969-12-23","end_date": "1970-01-23"},[
+            {"respondent_id": 1, "date": "1970-01-02", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 1, "date": "1970-01-09", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 1, "date": "1970-01-16", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-09", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-16", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 2, "answer": "Generous Giver"},
+            ])
+            ]
+
+
+
+@pytest.mark.parametrize("date_range,expected_response", TEST_DATA)
+def test_data_between_dateranges(date_range,expected_response):
+    "Test the response data recieved for different date ranges"
+    
+    #Get the response from API
+    data_date_range = date_range
+    json_data = json.dumps(data_date_range)
+    response = requests.post(responses_url, data = json_data, headers = {'User': API_KEY})
```

Comment:
Do we need lines 87, 88? Can't we just `data = json.dumps(date_range)` over here.

Reply:
fixed